iptables/iptables.go

```diff
@@ -94,8 +93,19 @@ func NewChain(name, bridge string, table Table, hairpinMode bool) (*Chain, error
 			return nil, fmt.Errorf("Could not create %s/%s chain: %s", c.Table, c.Name, output)
 		}
 	}
+	return c, nil
+}
 
-	switch table {
+// ProgramChain is used to add rules to a chain
+func ProgramChain(c *ChainInfo, hairpinMode bool) (*ChainInfo, error) {
```

Comment:
The signature of the api suggests that we accept a `*ChainInfo` and return a `*ChainInfo`. Why not just make this a Receiver method with the receiver type as `*ChainInfo`?